robotoff/prediction/ocr/category_from_AOC.py

```diff
@@ -0,0 +1,109 @@
+import re
+from typing import List, Optional, Union
+
+from robotoff.off import normalizing
+from robotoff.prediction.types import Prediction, PredictionType
+from robotoff.taxonomy import get_taxonomy
+from robotoff.utils import get_logger
+
+from .dataclass import OCRField, OCRRegex, OCRResult, get_text
+
+logger = get_logger(__name__)
+
+
+def category_taxonomisation(lang, match) -> Optional[str]:
+    """Function to match categories detected via AOP REGEX with categories
+    taxonomy database. If no match is possible, we return None."""
+
+    unchecked_category = lang + normalizing(match.group("category"))
+
+    checked_category = get_taxonomy("category").nodes.get(unchecked_category)
```

Comment:
We may add a TODO comment, because this way I think we are not trying to match with synonyms.
(for that we may have to create a utility function in Taxonomy).

Reply:
ping